rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,149 @@
+![cdevent-produce-webhook.png](cdevent-produce-webhook.png)
+
+The CDEvent will be received by a `events-broker` configured as Webhook URL.
+
+With this existing Webhook configuration CDEvents will not be fulfilled as additional CI/CD specific extensions can not be added here. This will be achieved by creating a new Notification type `CDEvents` to produce CDEvents using [Java SDK](https://github.com/cdevents/sdk-java) as part of Goal-2 Implementation.
+
+### Goal-2 : 
+Integrating CDEvents to Spinnaker using Spinnaker Pub/Sub Automated triggers type(google/amazon) and Notifications:
+
+### Consume CDEvents:
+
+CDEvents can be consumed by existing Pub/Sub system types(google/amazon) as CDEvent is build upon CloudEvent and CloudEvent is supported by most of the event providers.
+
+The pipeline can be triggered when a specific event type is received, event type will be included as `ce-type` in the HTTP Header part of the event consumed and it will be configured as one of the Attribute Constraints
+
+![cdevent-consume-pubsub.png](cdevent-consume-pubsub.png)
+
+
+#### Produce CDEvents
+A new Notification type `CDEvents` will be implemented to produce CDEvents using [Java SDK](https://github.com/cdevents/sdk-java).
+
+A Notification will be added at any stage of the pipeline from which we need to send a CDEvent.
+A CDEvent Notification will be sent to the selected `events-broker` and the event type configured when adding new Notification
+
+`CDEvents` can be created using [Java SDK](https://github.com/cdevents/sdk-java) that is available in `CDEvents` [GitHub Repositories](https://github.com/cdevents).
+`Java SDK` used to create various types of events aligning with `CDEvents` vocabulary specification using CloudEventBuilder. 
+
+`CDEvents` can be published to `events-broker` once created from Spinnaker application using CloudEvents HTTP libraries.
+
+Example: A sample service-deployed event published to `events-broker`
+```
+curl -v -d '{"id": "1234", "subject": "event"}' -X POST -H "Ce-Id: HelloSpinnaker" -H "Ce-Specversion: 1.0" -H "Ce-Type: dev.cdevents.service.deployed" -H "Ce-Source: not-sendoff" -H "Content-Type: application/json" "http://<events-broker-url>/default/events-broker"
+```
+
+### Dependencies
+
+The CloudEvent dependencies(io.cloudevents) are required to implement produce/consume events 
+
+## Drawbacks
+
+
+
+## Prior Art and Alternatives
+The existing approach is integration oriented and could sometimes result in outages due to change in implementation of integrated components. `CDEvents` allows users to take a more standardised approach, ensuring efficiency and sustainability while creating and maintaining CI/CD pipelines.
+
+## Known Unknowns
+
+The approach will be demonstrated with the Spinnaker Community using `CDEvents` PoC.
+
+## Security, Privacy, and Compliance
+
+
+## Operations
+Admin needs to configure Pipelines that should run on consuming the different event types from `events-broker`
```

Comment:
Can we go in depth? This is really not saying much. Can we have an example here showing the changes needed for a single API for example?